test/tests/functional/pbs_reservations.py

```diff
@@ -42,6 +42,226 @@
+        self.server.manager(MGR_CMD_SET, NODE, a, nodes[1]['id'])
+        self.__class__._nodes_configured = True
+
+    def submit_resv(self):
+        r = Reservation(TEST_USER)
+        a = {'Resource_List.select': '2:ncpus=1'}
+        r.set_attributes(a)
+        rid = self.server.submit(r)
+        try:
+            a = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+            d = self.server.expect(RESV, a, rid)
+        except PtlExpectError, e:
+            d = e.rv
+        return d
+
+    def test_no_span_psets(self):
+        self.create_nodes()
+        a = {'do_not_span_psets': '1'}
+        self.server.manager(MGR_CMD_SET, SCHED, a, "")
+        d = self.submit_resv()
+        self.assertTrue(d)
+
+    def test_over_limit(self):
+        self.create_nodes()
+        a = {'max_run_res.ncpus': '[u:' + str(TEST_USER) + '=1]'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        d = self.submit_resv()
+        self.assertTrue(d)
+
+    def test_resv_and_job(self):
+        self.create_nodes()
+        a = {'max_run_res.ncpus': '[u:' + str(TEST_USER) + '=3]'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        d = self.submit_resv()
+        self.assertTrue(d)
+        j = Job(TEST_USER)
+        j.set_attributes({'Resource_List.walltime': 37 * 3600})
+        jid = self.server.submit(j)
+        a = {'job_state': 'R'}
+        self.server.expect(JOB, a, jid)
+
+    def test_round_robin_reservation(self):
+        a = {'resources_available.ncpus': 4}
+        self.server.create_vnodes('vnode', a, 3, self.mom,
+                                  usenatvnode=True)
+
+        self.scheduler.sched_config[
+            'node_sort_key'] = "\"ncpus HIGH unused\" ALL"
+        self.scheduler.apply_config()
+        r1 = Reservation(TEST_USER)
+        r1.set_attributes({'reserve_start': int(time.time() + 61)})
+        r1id = self.server.submit(r1)
+        r1id = r1id.split()[0]
+
+        a = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
+        self.server.expect(RESV, a, id=r1id)
+
+        r2 = Reservation(TEST_USER)
+        r2.set_attributes({'reserve_start': int(time.time() + 121)})
+        r2id = self.server.submit(r2)
+        r2id = r2id.split()[0]
+        a = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
+        self.server.expect(RESV, a, id=r2id)
+
+    def submit_standing_reservation(self, user, select, rrule, start, end):
+        """
+        helper method to submit a standing reservation
+        """
+        if 'PBS_TZID' in self.conf:
+            tzone = self.conf['PBS_TZID']
+        elif 'PBS_TZID' in os.environ:
+            tzone = os.environ['PBS_TZID']
+        else:
+            self.logger.info('Missing timezone, using America/Los_Angeles')
+            tzone = 'America/Los_Angeles'
+
+        a = {'Resource_List.select': select,
+             ATTR_resv_rrule: rrule,
+             ATTR_resv_timezone: tzone,
+             ATTR_resv_standing: '1',
```

Comment:
Is this required?  Won't the fact that there is an rrule be enough to say this is a standing reservation?